ddtrace/ddtrace.go

```diff
@@ -75,6 +75,10 @@ type Span interface {
 
 	// Context returns the SpanContext of this Span.
 	Context() SpanContext
+
+	// AddLink sets a casaul relatationship between one or more spans
+	LinkSpan(spanContext SpanContext, attributes map[string]string)
+	AddLink(link SpanLink)
```

Comment:
We can't do this without breaking backwards compatibility, since extending an interface would break old versions of the code with interface implementors that don't have these methods. Let's not change the Datadog API. Just scope this to the OTel API for now.

Fine to keep the `WithSpanLinks` function that returns a `StartSpanOption` though.

Reply:
@dianashevchenko can you give some context about the motivation for this, since it came from your PR?

Reply:
According to the specification we need a way to add links to a span at any point. Removing these methods would restrict us to only adding links when a span is started.

Reply:
discussed this offline: This is breaking change and will require a major version release. We will punt on this requirement.